integration_tests/testsuite/test_custom.py

```diff
@@ -39,19 +54,140 @@ def runTest(self):
         if not output:
             logging.debug('No custom tests specified.')
         else:
-            for test_info in json.loads(output):
+            for specification in json.loads(output):
                 test_num += 1
-                name = test_info.get('name', 'test_{0}'.format(test_num))
-                path = test_info.get('path')
-                if path is None:
-                    logging.warn('Test \'%s\' has no path specified! '
-                                 'Skipping...', name)
-                    continue
+                self._runTestForSpecification(specification, test_num)
+
+    def _runTestForSpecification(self, specification, test_num):
+        """ Given the specification for a test execute the steps and
+            validate the result.
+
+        :param specification: Dictionary containing the specification.
+        :param test_num: Identifier of the test.
+        :return: None.
+        """
+        name = specification.get('name', 'test_{0}'.format(test_num))
+        timeout = specification.get('timeout', 500)
+        path = specification.get('path')
+        steps = specification.get('steps')
+        validation = specification.get('validation')
+
+        logging.info('Running custom test: %s', name)
+
+        if path is not None:
```

Comment:
Could you add a comment here or factor this into a sub-function to make it clear that this is the "legacy" path?